tests/aggregation_regress/tests.py

```diff
@@ -360,6 +361,40 @@ def test_aggrate_annotation(self):
             {'c__max': 3}
         )
 
+    def test_conditional_aggregate_of_subquery(self):
+        """
+        Regression test for #25307: AttributeError: 'WhereNode' object has no
+        attribute 'get_source_expressions'.
+
+        Tests conditional aggregations over subqueries.
+        """
+        # Conditional aggregation on a limited queryset.
```

Comment:
I don't think you need to test this case as it's used to denote the existence of `group_by` or not (it defaults to `None`). From what I understand this instance check could be replaced by a simple boolean check `if self.group_by ...`.